[#156753589] Add tasks to the billable events

paroxp opened the pull request:
## What

Currently, the tasks are being ignored when it comes to converting data
into our defined standard. Tasks too cost money, therefore we'd like to
make sure those are standardised and counted towards any billing
calculations.

## How to review

- Run tests
- Run application against your environment
- Create and kill a task
- Request and see data (tested myself)

dcarley commented on eventstore/sql/create_events.sql:
```diff
@@ -66,6 +66,30 @@ INSERT INTO events with
+				(raw_message->>'org_guid')::uuid as org_guid,
+				(raw_message->>'space_guid')::uuid as space_guid,
+				'ebfa9453-ef66-450c-8c37-d53dfd931038'::uuid as plan_guid,  -- plan guid for all compute resources
+				'task'::text as plan_name,                                  -- plan name for all compute resources
```
Should these comments say `for all task resources`?

paroxp replied:
Good catch :D

paroxp commented on eventstore/store_billable_events_test.go:
```diff
@@ -110,6 +110,95 @@ var _ = Describe("GetBillableEvents", func() {
+			Name:      "PLAN1",
+			Components: []eventstore.PricingPlanComponent{
+				{
+					Name:         "compute",
```
Yup. Shame the tests didn't blow due to that... Will change.

dcarley replied:
They have now 😆 (Travis build failed because the name appears in the assertion too)

paroxp commented:
I've addressed the comments.

Regarding the additional tests, I too felt it would be unnecessary to duplicate the tests. I realise there could be edge cases.

I also didn't think a task would run for longer enough to start before and stop after the requested time period.

Not sure what the solution is. I'm happy to duplicate the other tests.

dcarley commented:
> I also didn't think a task would run for longer enough to start before and stop after the requested time period.

Yeah, good point, because we're reporting on months. Starting before or ending after would be quite common though.

Maybe we should talk to @chrisfarms since he recently refactored all of the tests and one possible solution to this would be to have more magic/meta tests that check the same behaviours of both apps and tasks?

dcarley commented:
I spoke to Chris. Something that I hadn't quite grokked before is that we're normalising the task states to `STARTED` and `STOPPED` which are meanings that are interpreted later in the processing when you might be missing one of those states. So it's probably fine to only test this scenario for tasks 👍 

Thanks for the fixes, merging 👍 

Do we need to release in paas-cf?
